api/layer/layer.go

```diff
@@ -822,6 +825,8 @@ func (n *layer) deleteObject(ctx context.Context, bkt *data.BucketInfo, obj *Ver
 		obj.DeleteMarkVersion = objInfo.Version()
 	}
 
+	obj.DeleteMarkerEtag = objInfo.HashSum
```

Comment:
We have to make sure that we create delete marker. The following cases must not lead to creating delete marker tag notification:
* we delete specific `versionId` (`null` version is included)
* we delete object in unversioned bucket

Reply:
I thought we create delete marker in any case:
```
Current implementation doesn't consider "unversioned" mode (so any deletion creates "delete-mark" object).
```
Also don't we create a delete marker here: 
```
objInfo, err := n.objectPut(ctx, bkt, p)
	if err != nil {
		obj.Error = err
		return obj
	}
```

I think we should send notifications according to implementation.

Reply:
Actually this comment means: we don't support `unversioned` mode. But `versioned` and `suspended` modes are supported. And using these mode we can delete whole object (with delete mark creation) and delete specific version (without delete mark creation)

> so any deletion creates "delete-mark" object

It's about inner representation. It shouldn't affect application logic - we can still delete specific version (and for caller this version disappears though new object is created in NeoFS)